rfd/0011-database-access.md

```diff
@@ -0,0 +1,698 @@
+> db3 Database 3  env=prod
+```
+
+To log out of a particular database (i.e. remove certificate from key store):
+
+```sh
+$ tsh db logout db1
+```
+
+### Selecting databases and database user
+
+The `tsh db login` command also prints a footer explaining how to connect
+to the database:
+
+```
+Connection information for "postgres" has been saved to ~/.pg_service.conf.
+You can connect to the database using the following command:
+
+  $ psql "service=postgres user=<user> dbname=<dbname>"
+
+Or configure environment variables and use regular CLI flags:
+
+  $ eval $(tsh db env)
+  $ psql -U <user> <database>
+```
+
+After a successful login, users can use any database user and database to
+connect to, as long as it is allowed by the RBAC rules. For example:
+
+```sh
+$ tsh db login postgres-prod && eval $(tsh db env)
+$ psql -U alice mydb     # allowed by role so will connect
+$ psql -U bob mydb       # allowed by role so will connect
+$ psql -U charlie mydb   # not allowed by role so will deny access
+```
+
+Default database user and name can also be provided to the `tsh db login`
+command, in which case then won't be required when using `psql` or other
+client:
+
+```sh
+$ tsh db login --db-user=alice --db-name=mydb postgres-prod
+$ psql "service=postgres-prod"
+$ eval $(tsh db env)
+$ psql
+```
+
+Users can still choose to specify a user/database explicitly when connecting,
+for example to use a different database user or connect to a different 
+database (subject to RBAC checks).
+
+### Connecting to PostgreSQL / service file
+
+After logging into a PostgreSQL database, `tsh` configures an appropriate
+section in the [connection service file](https://www.postgresql.org/docs/9.1/libpq-pgservice.html)
+which is located at `~/.pg_service.conf`.
+
+The service file has ini format and can contain multiple sections where each
+section defines connection parameters for a particular "service". PostgreSQL
+clients can refer to a particular service using `service` connection string
+parameter which all libpq-based clients should recognize (e.g. `psql` and
+`pgAdmin` do).
+
+The section is named after the Teleport database service instance a user has
+logged into with `tsh db login <name>` command. If a service file is already
+present, a new section will be added or existing section with the same name
+overwritten. 
+
+An added section may look like this:
+
+```ini
+[postgres-prod]
+host=proxy.teleport.example.com
+port=3080
+sslmode=verify-full
+sslrootcert=/home/user/.tsh/keys/proxy.teleport.example.com/certs.pem
+sslcert=/home/user/.tsh/keys/proxy.teleport.example.com/alice-x509.pem
+sslkey=/home/user/.tsh/keys/proxy.teleport.example.com/alice
+```
+
+Users can then connect to the database using the following command:
+
+```sh
+$ psql "service=postgres-prod user=alice dbname=metrics"
+```
+
+Alternatively, `tsh` can output a set of environment variables supported by
+PostgreSQL clients for users to set in their session:
+
+```sh
+$ tsh db env
```

Comment:
optional, but I'd really appreciate if this wasn't always bash-specific.
e.g. I use fish, and the syntax is `set -x FOO bar` instead of `export FOO=bar`.
the shell can be detected via `$SHELL`.

Reply:
Good call, I can see it being useful. I'll create a follow up ticket for this.